### DISCUSSION TITLE
Bot disconnects after short time after join

### DISCUSSION THREAD

**Discussion by Jurogas3** [PrismarineJS/mineflayer#2428] [Q&A] [answered]
So, I made a bot - just for fun - and everytime I try to run him he runs for around 5 secs.
After that he throws this error:
Error: write ECONNRESET
    at afterWriteDispatched (node:internal/stream_base_commons:164:15)
    at writeGeneric (node:internal/stream_base_commons:155:3)
    at Socket._writeGeneric (node:net:795:11)
    at Socket._write (node:net:807:8)
    at writeOrBuffer (node:internal/streams/writable:389:12)
    at _write (node:internal/streams/writable:330:10)
    at Socket.Writable.write (node:internal/streams/writable:334:10)
    at Cipheriv.ondata (node:internal/streams/readable:754:22)
    at Cipheriv.emit (node:events:390:28)
    at addChunk (node:internal/streams/readable:315:12) {
  errno: -4077,
  code: 'ECONNRESET',
  syscall: 'write'
}

Can someone help me?

**Comment by amoraschi** [comment]
Provide more info, what's your code

**Reply by Jurogas3** [reply]
> Provide more info, what's your code

Hello, sorry for my late reply.

My Code is simple: 
const bot = mineflayer.createBot({
        host: '', 
        username: user,
        password: pass
    });
    const mcData = require('minecraft-data')(bot.version);

    bot.on('error', (err) => console.log(err));

    bot.on('login', () => {
        console.log(`[${bot.username}] Logged in`);
        bot.chatAddPattern(/(\w*) wants to teleport to you./, "tpRequest");
        bot.chatAddPattern(/You cannot talk until you have moved!/, "moveToTalk");
    });

    bot.loadPlugin(pathfinder);

    bot.on('spawn', () => {
        const mcData = require('minecraft-data')(bot.version);
        const defaultMove = new Movements(bot, mcData);
        bot.pathfinder.setMovements(defaultMove)
        bot.pathfinder.setGoal(new GoalNear(0, 4, 0, 1));
    });

    bot.on('chat', (username, message) => {
        if (username === bot.username) return
    });

    bot.on('chat', (username, message) => {
        if(bot.username == username) return;
        console.log(`[${bot.username}] ${username}: ${message}`);
        switch(message){
            case '!kill ' + bot.username:
                bot.chat('/kill');
        }
    });

    bot.on('tpRequest', function (username) {
        bot.chat(`/tpy ${username}`);
        bot.chat(`/msg ${username} Welcome to Spawn!`);
    });

    bot.on('moveToTalk', function () {
        
    });

    bot.once('spawn', () => {
    });

I even tried it on google console, but same problem.

**Comment by amoraschi** [comment, marked as answer]
The host options could be wrong, or move bot.loadPlugin and everything inside the login event, inside the spawn event

**Reply by Jurogas3** [reply]
Hello, I figured out that this part: 
`bot.loadPlugin(pathfinder);

bot.on('spawn', () => {
    const mcData = require('minecraft-data')(bot.version);
    const defaultMove = new Movements(bot, mcData);
    bot.pathfinder.setMovements(defaultMove)
    bot.pathfinder.setGoal(new GoalNear(0, 4, 0, 1));
});`

causes the error and timeout.
I just try to adjust the code, thanks for your reply!
